tests/Feature/Api/Users/UpdateUserApiTest.php

```diff
@@ -58,4 +59,68 @@ public function testApiUsersCanBeDeactivatedWithBooleanFalse()
 
         $this->assertEquals(0, $user->refresh()->activated);
     }
+
+    public function testUsersScopedToCompanyDuringUpdateWhenMultipleFullCompanySupportEnabled()
+    {
+        $this->settings->enableMultipleFullCompanySupport();
+
+        $companyA = Company::factory()->create(['name'=>'Company A']);
+        $companyB = Company::factory()->create(['name'=>'Company B']);
+
+        $adminA = User::factory(['company_id' => $companyA->id])->admin()->create();
+        $adminB = User::factory(['company_id' => $companyB->id])->admin()->create();
+        $adminNoCompany = User::factory(['company_id' => null])->admin()->create();
+
+        // Create a user that belongs to company B
+        $userA = User::factory()->create(['activated' => true, 'company_id' => $companyA->id]);
+        $userB = User::factory()->create(['activated' => true, 'company_id' => $companyB->id]);
+        $userNoCompany = User::factory()->create(['activated' => true, 'company_id' => null]);
```

Comment:
The comment here isn't correct and can go away if the variable name is more descriptive (and the rest of the test becomes clearer too):

```suggestion
		$userInCompanyA = User::factory()->create(['company_id' => $companyA->id]);
        $userInCompanyB = User::factory()->create(['company_id' => $companyB->id]);
        $userNoCompany = User::factory()->create(['company_id' => null]);
```

(`'activated' => true` is in the user factory's definition and is so core it probably won't be removed so I think it can be removed here)